webapp/channels/src/components/admin_console/access_control/editors/cel_editor/editor.tsx

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import * as monaco from 'monaco-editor';
+import React, {useCallback, useEffect, useRef, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import type {AccessControlTestResult} from '@mattermost/types/access_control';
+
+import {searchUsersForExpression} from 'mattermost-redux/actions/access_control';
+import {Client4} from 'mattermost-redux/client';
+
+import {MonacoLanguageProvider} from './language_provider';
+
+import CELHelpModal from '../../modals/cel_help/cel_help_modal';
+import TestResultsModal from '../../modals/policy_test/test_modal';
+import {TestButton, HelpText} from '../shared';
+
+import './editor.scss';
+
+export const POLICY_LANGUAGE = 'expressionLanguage';
+const VALIDATE_POLICY_SYNTAX_COMMAND_ID = 'policyEditorValidateSyntaxCommand';
+
+const MONACO_EDITOR_OPTIONS: monaco.editor.IStandaloneEditorConstructionOptions = {
```

Comment:
(Post-merge) (0/5) I know Monaco is cool, but given how much we're having to disable for this, how much of it are we actually using? Are we just using it for the autocomplete? It seems like it might have ended up being a bit overkill for what we need it for here.

As an alternative, we could possibly use `SuggestionBox`, but I'm a bit terrified of trying to use that here given how janky it can already be 😅